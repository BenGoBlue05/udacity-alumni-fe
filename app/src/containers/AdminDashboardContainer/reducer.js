import * as types from './constants';

export const initialState = {
  activeTab: 0,
  users: {
    editing: null,
    items: [],
    perPage: 10,
    currentPage: 1,
  },
  articles: {
    items: [],
    perPage: 10,
    currentPage: 1,
  },
  aside: {
    isVisible: false,
  },
  error: null,
  userTable: {
    sortIndex: 1,
    sortAscending: true,
  },
  modal: {
    userId: null,
    avatarInput: null,
    isVisible: false,
  },
};

const adminDashboardReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.ADMIN_DASHBOARD_OPEN_MODAL:
        return {
          ...state,
          modal: {
            ...state.modal,
            isVisible: true,
            userId: action.userId,
          },
        };
      case types.ADMIN_DASHBOARD_CLOSE_MODAL:
        return {
          ...state,
          modal: {
            ...state.modal,
            isVisible: false,
          },
        };
      case types.ADMIN_DASHBOARD_EDIT_AVATAR_INPUT:
        return {
          ...state,
          modal: {
            ...state.modal,
            avatarInput: action.input,
          },
        };
      case types.ADMIN_DASHBOARD_SET_SORT_OPTIONS:
        return {
          ...state,
          userTable: {
            ...state.userTable,
            sortIndex: action.index,
            sortAscending: action.ascending,
          },
        };
      case types.ADMIN_DASHBOARD_SET_ERROR:
        return {
          ...state,
          error: action.error,
        };
      case types.ADMIN_DASHBOARD_CLEAR_ERROR:
        return {
          ...state,
          error: null,
        };
      case types.ADMIN_DASHBOARD_TOGGLE_ASIDE:
        return {
          ...state,
          aside: {
            isVisible: !state.aside.isVisible,
          },
        };
      case types.ADMIN_DASHBOARD_CLEAR_USER_EDITING:
        return {
          ...state,
          users: {
            ...state.users,
            editing: null,
          },
        };
      case types.ADMIN_DASHBOARD_SET_USER_EDITING:
        return {
          ...state,
          users: {
            ...state.users,
            editing: action.index,
          },
        };
      case types.ADMIN_DASHBOARD_SET_USERS_PAGE:
        return {
          ...state,
          users: {
            ...state.users,
            currentPage: action.page,
          },
        };
      case types.ADMIN_DASHBOARD_SET_ARTICLES_PAGE:
        return {
          ...state,
          articles: {
            ...state.articles,
            currentPage: action.page,
          },
        };
      case types.ADMIN_DASHBOARD_SET_USERS:
        return {
          ...state,
          users: {
            ...state.users,
            items: action.users,
          },
        };
      case types.ADMIN_DASHBOARD_SET_ARTICLES:
        return {
          ...state,
          articles: {
            ...state.articles,
            items: action.articles,
          },
        };
      case types.ADMIN_DASHBOARD_SET_ACTIVE_TAB:
        return {
          ...state,
          activeTab: action.tab,
        };
      default:
        return state;
    }
  };

export default adminDashboardReducer;
