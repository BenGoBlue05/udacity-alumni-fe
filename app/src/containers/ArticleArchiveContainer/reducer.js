import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  tag: null,
};

const articleArchiveReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.ARCHIVE_SET_TAG:
        return update(state, {
          tag: {
            $set: action.tag,
          },
        });
      default:
        return state;
    }
  };

export default articleArchiveReducer;
