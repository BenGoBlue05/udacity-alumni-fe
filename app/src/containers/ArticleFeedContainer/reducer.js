import * as types from './constants';

export const initialState = {
  // Initial State goes here!
};

const articleFeedReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.DEFAULT_ACTION:
        return state;
      default:
        return state;
    }
  };

export default articleFeedReducer;
