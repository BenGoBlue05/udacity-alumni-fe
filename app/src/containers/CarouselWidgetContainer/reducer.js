import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  currentlyEditing: null,
  images: [
    {
      id: 1,
      url:"https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/Udacity-Self-Driving-Car-Engineer-Nanodegree-800x333.jpg?raw=true",
    },
    {
      id: 2,
      url: "https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/VRDeveloperNanodegreeProgram.png?raw=true",
    },
    {
      id: 3,
      url: "https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/deeplearning.jpg?raw=true",
    },
  ],
};

const carouselWidgetReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.CAROUSEL_SET_EDITING:
        return update(state, {
          currentlyEditing: {
            $set: action.index,
          },
        });
      case types.CAROUSEL_CANCEL_EDITING:
        return update(state, {
          currentlyEditing: {
            $set: null,
          },
        });
      case types.CAROUSEL_ADD_IMAGE:
        return update(state, {
          images: {
            $set: [...state.images, action.image],
          },
        });
      case types.CAROUSEL_REMOVE_IMAGE:
        return update(state, {
          images: {
            $set: [
              ...state.images.slice(0, action.index),
              ...state.images.slice(action.index + 1),
            ],
          },
        });
      case types.CAROUSEL_EDIT_IMAGE:
        return update(state, {
          images: {
            $set: [
              ...state.images.slice(0, action.index),
              action.image,
              ...state.images.slice(action.index + 1),
            ],
          },
          currentlyEditing: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default carouselWidgetReducer;
