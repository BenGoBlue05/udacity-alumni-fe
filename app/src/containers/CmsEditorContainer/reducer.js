import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  article: {
    id: null,
    action: null,
  },
  editorState: null,
  editorTitle: null,
  preview: {
    isPreviewing: false,
    content: null,
    title: null,
  },
  isValid: false,
  isSubmitting: false,
  error: null,
  message: null,
  modal: {
    isShowing: false,
    status: 0,
    spotlighted: false,
    selectedTags: [],
  },
};

const previewReducer = (state = initialState.preview, action) => {
  switch (action.type) {
    case types.CMS_CLOSE_PREVIEW:
      return update(state, {
        isPreviewing: {
          $set: false,
        },
        markdown: {
          $set: null,
        },
        title: {
          $set: null,
        },
      });
    case types.CMS_SET_PREVIEW_STATE:
      return update(state, {
        isPreviewing: {
          $set: true,
        },
        content: {
          $set: action.markdown,
        },
        title: {
          $set: action.title,
        },
      });
    default: return state;
  }
};

const modalReducer = (state = initialState.modal, action) => {
  switch (action.type) {
    case types.CMS_TOGGLE_SPOTLIGHT:
      return update(state, {
        spotlighted: {
          $set: !state.spotlighted,
        },
      });
    case types.CMS_OPEN_MODAL:
      return update(state, {
        isShowing: {
          $set: true,
        },
      });
    case types.CMS_CLOSE_MODAL:
      return update(state, {
        isShowing: {
          $set: false,
        },
      });
    case types.CMS_SET_STATUS:
      return update(state, {
        status: {
          $set: action.status,
        },
      });
    case types.CMS_SET_SELECTED_TAGS:
      return update(state, {
        selectedTags: {
          $set: action.tags,
        },
      });
    default: return state;
  }
};

const cmsEditorReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.SUBMIT_ARTICLE_INITIATION:
        return update(state, {
          isSubmitting: {
            $set: true,
          },
        });
      case types.SUBMIT_ARTICLE_FAILURE:
        return update(state, {
          isSubmitting: {
            $set: false,
          },
          error: {
            $set: action.error,
          },
        });
      case types.SUBMIT_ARTICLE_SUCCESS:
        return update(state, {
          isSubmitting: {
            $set: false,
          },
          message: {
            $set: action.message,
          },
        });
      case types.CLEAR_CMS_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.CLEAR_CMS_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      case types.CMS_OPEN_MODAL:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_CLOSE_MODAL:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_STATUS:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_SELECTED_TAGS:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_TOGGLE_SPOTLIGHT:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_EDITOR_TITLE:
        return update(state, {
          editorTitle: {
            $set: action.title,
          },
          isValid: {
            $set: action.title !== '',
          },
        });
      case types.CMS_SET_EDITOR_STATE:
        return update(state, {
          editorState: {
            $set: action.state,
          },
        });
      case types.CMS_CLOSE_PREVIEW:
        return update(state, {
          preview: {
            $set: previewReducer(state.preview, action),
          },
        });
      case types.CMS_SET_PREVIEW_STATE:
        return update(state, {
          preview: {
            $set: previewReducer(state.preview, action),
          },
        });
      case types.CMS_SET_ARTICLE:
        return update(state, {
          article: {
            id: {
              $set: action.id,
            },
            action: {
              $set: action.action,
            },
          },
        });
      default:
        return state;
    }
  };

export default cmsEditorReducer;
