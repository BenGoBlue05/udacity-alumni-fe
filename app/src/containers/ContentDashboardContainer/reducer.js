import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  articles: null,
  error: null,
  isLoading: false,
};

const contentDashboardReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.DASHBOARD_ARTICLES_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.DASHBOARD_ARTICLES_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
          articles: {
            $set: action.articles,
          },
        });
      case types.DASHBOARD_ARTICLES_FAILURE:
        return update(state, {
          isLoading: {
            $set: false,
          },
          error: {
            $set: action.error,
          },
        });
      case types.CLEAR_DASHBOARD_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default contentDashboardReducer;
