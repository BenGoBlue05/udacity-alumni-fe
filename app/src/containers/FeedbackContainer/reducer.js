import * as types from './constants';

export const initialState = {
  feedbackModalOpen: false,
};

const feedbackReducer =
(state = initialState, action) => {
  switch (action.type) {
    case types.OPEN_FEEDBACK_MODAL:
      return {
        ...state,
        feedbackModalOpen: true,
      };
    default:
      return state;
  }
};

export default feedbackReducer;
