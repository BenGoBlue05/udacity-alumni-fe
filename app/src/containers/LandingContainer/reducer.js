import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  isLoading: false,
  carouselImages: [
    {
      src: 'http://zdnet4.cbsistatic.com/hub/i/r/2016/01/13/df0444a8-0bcd-46cc-b9d4-e5be56a36341/resize/770xauto/cd213f8602843b807ed63aef891203c7/image.jpg',
    },
    {
      src: 'http://zdnet4.cbsistatic.com/hub/i/r/2016/01/13/df0444a8-0bcd-46cc-b9d4-e5be56a36341/resize/770xauto/cd213f8602843b807ed63aef891203c7/image.jpg',
    },
    {
      src: 'http://zdnet4.cbsistatic.com/hub/i/r/2016/01/13/df0444a8-0bcd-46cc-b9d4-e5be56a36341/resize/770xauto/cd213f8602843b807ed63aef891203c7/image.jpg',
    },
  ],
  featuredArticles: null,
  errors: null,
};

const landingReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.FEATURED_ARTICLES_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.FEATURED_ARTICLES_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
          featuredArticles: {
            $set: action.articles,
          },
        });
      case types.FEATURE_ARTICLES_FAILURE:
        return update(state, {
          isLoading: {
            $set: false,
          },
          errors: {
            $set: action.errors,
          },
        });
      case types.CLEAR_LANDING_ERRORS:
        return update(state, {
          errors: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default landingReducer;
