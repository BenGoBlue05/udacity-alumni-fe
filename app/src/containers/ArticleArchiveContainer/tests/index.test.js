import ArticleArchive from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as archiveContainer } from '../reducer';
import { initialState as app } from '../../AppContainer/reducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const articles = [{
  id: 4,
  title: 'Current procedural terminology Benefit Impaired risk',
  featured: false,
  spotlighted: false,
  content: 'Diagnostic related group HIPAA accreditation',
  feature_image: 'https://robohash.org/sapienteeaoptio.png?size=300x300',
  status: 'published',
  slug: 'current-procedural-terminology-benefit-impaired-risk',
  user: {
    id: 3,
    email: 'andreas@gmail.com',
    created_at: '2016-09-30T02:39:57.297Z',
    updated_at: '2016-10-03T18:49:42.965Z',
    name: 'Andreas Daimainger',
    avatar: 'https://avatars3.githubusercontent.com/u/13679375?v=3&s=466',
    bio: 'Hello world',
    role: 'user',
  },
  created_at: '2016-09-30T02:39:57.352Z',
  updated_at: '2016-09-30T02:39:57.352Z',
}];

const loadingArticles = true;

describe('<ArticleArchive />', () => {
  it('should render with default props', () => {
  	const tag = 'Hello';
    const store = mockStore({ archiveContainer, app });
    const wrapper = shallow(
      <ArticleArchive 
        store={store}
	      selectedTag={tag} 
	      articles={articles} 
	      loadingArticles={loadingArticles} 
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
