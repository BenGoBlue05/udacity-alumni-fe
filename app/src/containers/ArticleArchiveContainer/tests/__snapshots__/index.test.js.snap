exports[`<ArticleArchive /> should render with default props 1`] = `
<Apollo(ArticleArchive)
  actions={
    Object {
      "setTag": [Function anonymous]
    }
  }
  articles={
    Array [
      Object {
        "content": "Diagnostic related group HIPAA accreditation",
        "created_at": "2016-09-30T02:39:57.352Z",
        "feature_image": "https://robohash.org/sapienteeaoptio.png?size=300x300",
        "featured": false,
        "id": 4,
        "slug": "current-procedural-terminology-benefit-impaired-risk",
        "spotlighted": false,
        "status": "published",
        "title": "Current procedural terminology Benefit Impaired risk",
        "updated_at": "2016-09-30T02:39:57.352Z",
        "user": Object {
          "avatar": "https://avatars3.githubusercontent.com/u/13679375?v=3&s=466",
          "bio": "Hello world",
          "created_at": "2016-09-30T02:39:57.297Z",
          "email": "andreas@gmail.com",
          "id": 3,
          "name": "Andreas Daimainger",
          "role": "user",
          "updated_at": "2016-10-03T18:49:42.965Z"
        }
      }
    ]
  }
  loadingArticles={true}
  selectedTag={null}
  store={
    Object {
      "clearActions": [Function clearActions],
      "dispatch": [Function anonymous],
      "getActions": [Function getActions],
      "getState": [Function getState],
      "replaceReducer": [Function replaceReducer],
      "subscribe": [Function subscribe]
    }
  } />
`;
